src/gui/sdlmain.cpp

```diff
@@ -4639,6 +4651,75 @@ void GFX_GetSize(int &width, int &height, bool &fullscreen)
 	fullscreen = sdl.desktop.fullscreen;
 }
 
+static bool check_env_for_high_dpi_handling()
+{
+	constexpr auto disable_high_dpi_var = "SDL_HINT_VIDEO_HIGHDPI_DISABLED";
+
+	const auto disable_high_dpi_val = getenv(disable_high_dpi_var);
+	if (disable_high_dpi_val == nullptr) {
+		return true;
+	}
+	if (std::string_view(disable_high_dpi_val) == "0" ||
+	    iequals(disable_high_dpi_val, "false") || iequals(disable_high_dpi_val, "no")) {
+		return true;
+	}
+
+	LOG_MSG("SDL: Ignoring DPI settings as requested by %s=%s environment variable",
+	        disable_high_dpi_var,
+	        disable_high_dpi_val);
+	return false;
+}
+
+static bool check_cli_for_dpi_handling(CommandLine& cmdline)
+{
+	for (const auto arg : {"--ignore-dpi", "-ignore-dpi"}) {
+		if (cmdline.FindExist(arg)) {
+			LOG_MSG("SDL: Ignoring DPI settings as requested by '%s' command-line argument",
+			        arg);
+			return false;
+		}
+	}
+	return true;
+}
+
+#if defined(WIN32)
+void static set_windows_dpi_awareness_hints(const bool want_dpi_handling)
+{
+#	define dpi_min_major_ver  2
+#	define dpi_min_minor_ver  23
+#	define dpi_min_bugfix_ver 0
+
+#	if SDL_VERSION_ATLEAST(dpi_min_major_ver, dpi_min_minor_ver, dpi_min_bugfix_ver)
+	if (want_dpi_handling) {
+		if (SDL_SetHint(SDL_HINT_WINDOWS_DPI_AWARENESS, "permonitorv2") == SDL_FALSE) {
+			LOG_WARNING("SDL: Failed to enable DPI awareness flag");
+		}
+		if (SDL_SetHint(SDL_HINT_WINDOWS_DPI_SCALING, "1") == SDL_FALSE) {
+			LOG_WARNING("SDL: Failed to enable DPI scaling flag");
+		}
+	} else {
+		if (SDL_SetHint(SDL_HINT_WINDOWS_DPI_AWARENESS, "unaware") == SDL_FALSE) {
+			LOG_WARNING("SDL: Failed to disable DPI awareness flag");
+		}
+		if (SDL_SetHint(SDL_HINT_WINDOWS_DPI_SCALING, "0") == SDL_FALSE) {
+			LOG_WARNING("SDL: Failed to disable DPI scaling flag");
```

Comment:
In the build that eXo tested (this PR), the `else` block was executed cleanly (no warnings) so we know it set the following hints:

```shell
set SDL_HINT_WINDOWS_DPI_AWARENESS="unaware"
set SDL_HINT_WINDOWS_DPI_SCALING="0"
```

These hints still allowed windows to apply it's 150% magnification:

<img width="930" alt="windows-150%-scale-windowresolution-800x600-ignore-dpi" src="https://user-images.githubusercontent.com/1557255/212141673-89833af0-1f28-48f3-a899-d5c24247c20e.PNG">

Where as v0.79.1 (and ECE) avoid the magnification all together.

Maybe instead of `unaware`, we as eXo to set the empty string, [which has different significance](https://github.com/libsdl-org/SDL/commit/51ebefeeee2228fe3a20e6985c4901d4e6a160d8)?

```shell
set SDL_HINT_WINDOWS_DPI_AWARENESS=""
set SDL_HINT_WINDOWS_DPI_SCALING="0"
```

Reply:
Once we've decided what to suggestion to eXo, I'll pass it to him via Discord. (and not using this PR build.. he'll try using v0.80.1)

Reply:
Explicitly declaring process as DPI unaware is asking Windows to scale the window, if I recall. I'd say, set just the `SDL_WINDOWS_DPI_SCALING` envvar to zero and leave the other hint alone. I _think_ Microsoft recommends declaring process as DPI aware when it needs access to all physical pixels (e.g. games) even if there's no actual DPI scaling done, but I don't have a source at hand.

Reply:
That did the trick!